src/dmd/ob.d

```diff
@@ -189,14 +191,19 @@ string toString(ObType obtype)
 
 enum PtrState : ubyte
 {
-    Undefined, Owner, Borrowed, Readonly
+    Initial, Undefined, Owner, Borrowed, Readonly
 }
 
 /************
  */
 const(char)* toChars(PtrState state)
 {
-    return ["Undefined", "Owner", "Borrowed", "Readonly"][state].ptr;
+    return toString(state).ptr;
+}
+
+string toString(PtrState state)
```

Comment:
Can you put `@nogc` on this to make sure it does not allocate ?

Reply:
Ok, but I'm not really thinking about attributes in ob.d at this stage.

Reply:
aannndddd.... that broke it with:
```
ERROR: dmd/ob.d(206): Error: array literal in @nogc function dmd.ob.toString may cause a GC allocation
```
so no, I'm not going to fight that battle at the moment.

Reply:
Exactly. That array literal is not const-folded, as I thought.
So instead, just use a manifest constant (`static immutable`) so we don't needlessly allocate on every call here. Then we can use the attribute soup on this function.

Reply:
I'm not concerned about attribute soup on ob.d at this time. This algorithm is a prototype and I've had to change it several times (like in this PR). I'm mostly concerned that the algorithm is correct. Several people have told me it can't be done :-)

Dealing with attributes, performance, and allocation efficiency comes later.

Reply:
It's a trivial change to use an manifest constant here. It's not about optimization, it's about not introducing a pattern which can be very easily avoided and is so evidently wasteful.